app/queries/district_analytics_query.rb

```diff
@@ -41,26 +41,12 @@ def registered_patients_by_period
   end
 
   def follow_up_patients_by_period
-    #
-    # this is similar to what the group_by_period query already gives us,
-    # however, groupdate does not allow us to use these "groups" in a where clause
-    # hence, we've replicated its grouping behaviour in order to remove the patients
-    # that were registered prior to the period bucket
-    #
-    date_truncate_string =
-      "(DATE_TRUNC('#{@period}', blood_pressures.recorded_at::timestamptz AT TIME ZONE '#{Groupdate.time_zone || 'Etc/UTC'}'))"
-
     @follow_up_patients_by_period ||=
-      BloodPressure
-        .left_outer_joins(:user)
-        .left_outer_joins(:patient)
-        .joins(:facility)
-        .where(facilities: { id: facilities })
-        .where(deleted_at: nil)
-        .group('facilities.id')
-        .group_by_period(@period, 'blood_pressures.recorded_at')
-        .where("patients.recorded_at < #{date_truncate_string}")
-        .order('facilities.id')
+      Patient
+        .joins(:blood_pressures)
+        .group('blood_pressures.facility_id')
+        .follow_ups(@period, last: @prev_periods)
+        .where(blood_pressures: { facility: facilities })
```

Comment:
This is the dashboard page that shows district cohort info, right?

Reply:
Yup.

Reply:
Ah yup, should have been clearer. Yes the tabular data that shown below cohort bars. 👍